### PR TITLE
Request to merge DB branch with the main branch

### PR DESCRIPTION
**Pull request by AleksandraZhi**
**The main page is ready (withoug database). 
Index.html changed into index.pug.
Css styling fixed.
Server setup and running.**

2020 09 15 chcat v2 00 header
2020 09 15 chcat v2 01 novels page
2020 09 15 chcat v2 02 shorts page
2020 09 15 chcat v2 03 inspired articles page
2020 09 15 chcat v2 04 haiku page
2020 09 15 chcat v2 05 footer

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const mongoose = require("mongoose");
+const routes = require("./routes/index");
+const helpers = require("./helpers");
+const path = require('path');
+
+const app = express();
+
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "pug");
+// app.set("vendors", path.join(__dirname, "../vendors"))
```
Comment (mforner13) [inline server/app.js]: Can this comment be removed now? :) 
It's best not to leave commented out code in your work as other developers who read it don't know why it's there.

**Review by mforner13 — COMMENTED** [review]
Before I review properly, I would recommend taking one of the following steps so that you don't add all of the bootstrap packages to git:
- Look up about [installing bootstrap via npm](https://www.npmjs.com/package/bootstrap) so that it is added to your node packages and automatically ignored from git
- Work with bootstrap CDN so that you can just link to the package rather than having to have everything installed
- Add your bootstrap folder to your `.gitignore` file and specify in your readme where someone can download it from in order to use it with the project

Let me know if you need any help! :)

File: server/app.js (hunk 1 above)
Comment (AleksandraZhi) [reply server/app.js]: Thank you for your review. 
Bootstrap: I've been struggling with bootstrap working with pug - it was the only way I could set it up. We can look into it on Wednesday. I had a jade-boostrap package installed, and now I tried installing just bootstrap npm - column styling still is lost with it. I had the same issue with the font-awesome, which I resolved the same way as bootstrap. I wonder if I've missed a "require"?
Commented out code: removed.
Thank you.

Comment (mforner13) [reply server/app.js]: No problem, let's talk about it on Monday. In the meantime, you can still git ignore the bootstrap folder which would at least make the rest of the PR easier to review :) Message me if you need any help with adding it to your git ignore file.

Comment (AleksandraZhi) [reply server/app.js]: Heya, the node_module, bootstrap, fontawesome folders were removed from the repo (removed-but-not-removed, added, double added, finally removed and history cleaned). I've used rebase and git gui to make the changes and pushed them into the git.
Awaiting your comments.

**Review by mforner13 — CHANGES_REQUESTED** [review]
Great job removing the bootstrap files!

Unfortunately it looks like the `fontawesome` library has snuck in though, so you may need to repeat what you did to remove bootstrap to remove these files. 

A good way of checking what's in the PR is the number of files or lines added in the top right corner. A PR shouldn't normally have more than 500ish lines of code in the green.

Screenshot 2020-10-01 at 16 31 25

**Comment by AleksandraZhi** [comment]
Looks like I overdid it with removing the files. Unfortunately, I didn't back up my project and while doing rebasing I removed a lot of files from both repo. 
The server is not running anymore. I will try to reverse to the stage before I started removing all the files:)

**Comment by mforner13** [comment]
> Looks like I overdid it with removing the files. Unfortunately, I didn't back up my project and while doing rebasing I removed a lot of files from both repo.
> The server is not running anymore. I will try to reverse to the stage before I started removing all the files:)

Oh no! If you run the command `git reflog` you will be able to see a timeline of absolutely everything you've done. You may be able to get some commit hashes from that if they haven't been overwritten by git yet? Let me know if you would like a call (when you're feeling better) to go through this!

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/server/node_modules
+/server/variables.env
+/server/public/books
+/server/public/bootstrap
+/server/public/fontawesome-free
```
Comment (mforner13) [inline .gitignore]: You always need a blank line at the end of files :) Github will always let you know when it's missing with this red circle

File: server/models/Books.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const mongoose = require("mongoose");
+mongoose.Promise = global.Promise;
```
Comment (mforner13) [inline server/models/Books.js]: It's good practice to have your imports grouped at the top, and then everything else after - e.g. this line :)

File: server/models/Books.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const slug = require("slugs");
+
+const novelSchema = new mongoose.Schema({
+  title: String,
+  slug: String,
+})
+
+const chapterSchema = new mongoose.Schema ({
+  //foreign key - need to research how to connect to an id in novelSchema - like novelID
+  // chaperNumber:
```
Comment (mforner13) [inline server/models/Books.js]: Good to keep the code you merge into master clean - if you're writing a reminder for something which needs to be added in a future PR, you can add `TODO` at the beginning - e.g. `// TODO: add chapterNumber once I've done xyz...`.

File: server/models/Books.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  chapterName: String,
+  text: String,
+  slug: String,
+})
+
+const storiesArticlesSchema = new mongoose.Schema ({
+  type: String, //create constants to have a choice wether story, article
```
Comment (mforner13) [inline server/models/Books.js]: Same here - prefix with `TODO` to make it clear what the purpose of the comment is.

File: server/routes/index.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+<<<<<<< HEAD
+// router.get("/", mainController.homePage);
+
+router.get('/', (req, res) => {
+  res.render('index');
+});
+=======
+router.get("/", mainController.novelsPage);
+>>>>>>> b00a69c... add templating
```
Comment (mforner13) [inline server/routes/index.js]: Looks like there's some left over symbols from a merge conflict which need to be removed. Also can line 6 be removed now?

File: server/views/index.pug [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+    link(rel='stylesheet', href='/css/main-styles.css')
+    link(rel='stylesheet', href='/bootstrap/dist/css/bootstrap.min.css')
+    link(rel='stylesheet', href='/fontawesome-free/css/all.css')
+    link(rel='image', href='/images/img-header-md.jpg')
+    //- link(rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous")
+    link(href="https://fonts.googleapis.com/css2?family=EB+Garamond:ital,wght@0,400;0,700;1,400;1,600&family=Inter:wght@100;200;300&display=swap" rel="stylesheet")
+    //- link(href="/../../vendors/css/js/all.js" rel="stylesheet")
+
+    //- style
+    //- include /images
```
Comment (mforner13) [inline server/views/index.pug]: Can these comments be removed? :)

**Review by mforner13 — COMMENTED** [review]
Looks great! :D Just a few comments which should be easy to fix.

File: server/views/index.pug (hunk 7 above)
Comment (AleksandraZhi) [reply server/views/index.pug]: removed

File: server/routes/index.js (hunk 6 above)
Comment (AleksandraZhi) [reply server/routes/index.js]: cleaned

File: server/models/Books.js (hunk 5 above)
Comment (AleksandraZhi) [reply server/models/Books.js]: removed

File: server/models/Books.js (hunk 4 above)
Comment (AleksandraZhi) [reply server/models/Books.js]: removed

File: .gitignore (hunk 2 above)
Comment (AleksandraZhi) [reply .gitignore]: added a blank line

File: server/models/Books.js (hunk 3 above)
Comment (AleksandraZhi) [reply server/models/Books.js]: regrouped

**Review by mforner13 — APPROVED** [review]
Looks great!! I'm sorry it took me so long to come back to it! There will be some merge conflicts when you merge this into master, so when this happens, you'll just need to choose the newer lines of code which you want to keep. Let me know if you want to go through this together :)

**Pull request merged**